cirq/ops/raw_types_test.py

```diff
@@ -459,6 +459,69 @@ def test_tagged_operation():
     assert op.with_qubits(q2).qubits == (q2,)
 
 
+def test_circuit_diagram():
+    h = cirq.H(cirq.GridQubit(1, 1))
+    tagged_h = h.with_tags('tag1')
+    expected = cirq.CircuitDiagramInfo(wire_symbols=("H['tag1']",),
+                                       exponent=1.0,
+                                       connected=True,
+                                       exponent_qubit_index=None,
+                                       auto_exponent_parens=True)
+    args = cirq.CircuitDiagramInfoArgs(None, None, None, None, None, False)
+    assert cirq.circuit_diagram_info(tagged_h) == expected
+    assert (cirq.circuit_diagram_info(tagged_h,
+                                      args) == cirq.circuit_diagram_info(h))
+
+    c = cirq.Circuit(tagged_h)
+    diagram_with_tags = "(1, 1): ───H['tag1']───"
+    diagram_without_tags = "(1, 1): ───H───"
+    assert str(cirq.Circuit(tagged_h)) == diagram_with_tags
+    assert c.to_text_diagram() == diagram_with_tags
+    assert c.to_text_diagram(include_tags=False) == diagram_without_tags
+
+
+def test_circuit_diagram_corner_cases():
+    # Tests global phase operation
+    global_phase = cirq.GlobalPhaseOperation(coefficient=1.0).with_tags('tag0')
+    expected_diagram = "\n\nglobal phase ['tag0']: 1.0"
+    assert global_phase._circuit_diagram_info_(args=None) == NotImplemented
+    assert cirq.Circuit(global_phase).to_text_diagram() == expected_diagram
+    expected = cirq.CircuitDiagramInfo(wire_symbols=tuple(),
+                                       exponent=1.0,
+                                       connected=True,
+                                       exponent_qubit_index=None,
+                                       auto_exponent_parens=True)
+
+    # Operation with no qubits and returns diagram info with no wire symbols
+    class NoWireSymbols(cirq.GlobalPhaseOperation):
```

Comment:
I don't know if this needs to be tested since non-global phase zero-qubit ops are not really supported anyways.

Reply:
Maybe not, but I wanted to make sure that it didn't raise an exception, since parts of the code assume that a wire symbol exists.  I'll keep it in for now.